api/app/controllers/gift_cards_controller.rb

```diff
@@ -1,8 +1,63 @@
+require 'stripe'
+
 class GiftCardsController < ApplicationController
+  before_action :set_gift_card, only: [:show, :update, :destroy]
 
   # POST /gift_cards
   def create
-    obj = { message: 'hello world!' }
-    json_response(obj)
+    @gift_card = GiftCard.create!(create_params)
+    json_response(@gift_card, :created)
+  end
+
+  # GET /gift_cards/:d
+  def show
+    json_response(@gift_card)
+  end
+
+  # PUT /gift_cards/:id
+  def update
```

Comment:
Update should also take an email and verify that the email is the same. It would need to pull the user from Stripe via the charge